lua/luasnip/nodes/functionNode.lua

```diff
@@ -9,16 +8,108 @@ local key_indexer = require("luasnip.nodes.key_indexer")
+---  `user_arg1`-`user_argn`.
+---
+---  These make it easier to reuse similar functions, for example a functionNode
+---  that wraps some text in different delimiters (`()`, `[]`, ...).
+---  ```lua
+---  local function reused_func(_,_, user_arg1)
+---      return user_arg1
+---  end
+---
+---  s("trig", {
+---      f(reused_func, {}, {
+---          user_args = {"text"}
+---      }),
+---      f(reused_func, {}, {
+---          user_args = {"different text"}
+---      }),
+---  })
+---  ```
+
+--- Function Nodes insert text based on the content of other nodes using a
+--- user-defined function:
+---
+--- ```lua
+--- local function fn(
+---   args,     -- text from i(2) in this example i.e. { { "456" } }
+---   parent,   -- parent snippet or parent node
+---   user_args -- user_args from opts.user_args
+--- )
+---    return '[' .. args[1][1] .. user_args .. ']'
+--- end
+---
+--- s("trig", {
+---   i(1), t '<-i(1) ',
+---   f(fn,  -- callback (args, parent, user_args) -> string
+---     {2}, -- node indice(s) whose text is passed to fn, i.e. i(2)
+---     { user_args = { "user_args_value" }} -- opts
+---   ),
+---   t ' i(2)->', i(2), t '<-i(2) i(0)->', i(0)
+--- })
+--- ```
+---
+---@param fn LuaSnip.FunctionNode.Fn
+---
+---  - `argnode_text`: The text currently contained in the argnodes
+---    (e.g. `{{line1}, {line1, line2}}`).
+---    The snippet indent will be removed from all lines following the first.
+---
+---  - `parent`: The immediate parent of the `functionNode`. It is included here
+---    as it allows easy access to some information that could be useful in
+---    functionNodes (see [Snippets-Data](#data) for some examples).
+---
+---    Many snippets access the surrounding snippet just as `parent`, but if the
+---    `functionNode` is nested within a `snippetNode`, the immediate parent is
+---    a `snippetNode`, not the surrounding snippet (only the surrounding
+---    snippet contains data like `env` or `captures`).
+---
+---  - `user_args`: The `user_args` passed in `opts`. Note that there may be
+---    multiple `user_args` (e.g. `user_args1, ..., user_argsn`).
+---
+---  The function shall return a string, which will be inserted as is, or a
+---  table of strings for multiline strings, where all lines following the first
+---  will be prefixed with the snippets' indentation.
+---
+---@param argsnode_refs? LuaSnip.NodeRef[]|LuaSnip.NodeRef
+---  [Node References](#node-reference) to the nodes the functionNode depends
+---  on.
+---  Changing any of these will trigger a re-evaluation of `fn`, and insertion of
+---  the updated text.
+---  If no node reference is passed, the `functionNode` is evaluated once upon
+---  expansion.
+---
+---@param node_opts? LuaSnip.Opts.FunctionNode
+---@return LuaSnip.FunctionNode
+--
+-- FIXME(@bew): The super flexible NodeRef param & the fact that the NodeRef
+-- type is an alias, makes luals make a huge function signature here 👀
```

Comment:
Settings `hover.expandAlias` in `lua-language-server` to false makes this work pretty well

<img width="995" height="68" alt="screen" src="https://github.com/user-attachments/assets/0e5a7acd-a137-464d-81bf-7cf7c63320a6" />

Honestly, setting that does not seem like a bad idea at all :eyes: The only annoying bit is that it doesn't seem like one can jump to definition from the documentation window :/
(and obviously that not everyone will want to set this only to make this annotation work :D)